File: src/codeforces/contest1339/ProblemC.java
package codeforces.contest1339;

import java.util.Scanner;

public class ProblemC {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        for (int i = 0; i < t; i++) {
            int n = scanner.nextInt();
            long[] a = new long[n];
            for (int j = 0; j < n; j++) {
                a[j] = scanner.nextInt();
            }
            int ans = 0;
            for (int j = 1; j < n; j++) {
                if (a[j] < a[j - 1]) {
                    if (a[j] + (1L << ans) - 1 < a[j - 1]) {
                        ans = 64 - Long.numberOfLeadingZeros(a[j - 1] - a[j]);
                    }
                    a[j] = a[j - 1];
                }
            }
            System.out.println(ans);
        }
    }
}

File: src/codeforces/contest1398/ProblemC.kt
@file:Suppress("DuplicatedCode")

package codeforces.contest1398

import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.util.*

fun main(args: Array<String>) {
    val tests = nextInt()
    for (t in 0 until tests) {
        val n = nextInt()
        val s = next()
        println(solve(s))
    }
}

fun solve(arr: String): Long {
    var ans = 0L
    val map = HashMap<Long, Int>()
    map[0L] = 1
    var curr = 0L
    for (num in arr) {
        curr += num - '0' - 1
        ans += map[curr] ?: 0
        map[curr] = (map[curr] ?: 0) + 1
    }
    return ans
}


private class MyReader(inputStream: InputStream) {
    private val reader = BufferedReader(InputStreamReader(inputStream))
    private var tokenizer: StringTokenizer? = null

    fun next(): String {
        while (true) {
            tokenizer.let {
                if (it == null || !it.hasMoreTokens()) {
                    tokenizer = StringTokenizer(reader.readLine())
                } else {
                    return it.nextToken()
                }
            }
        }
    }
}

private val reader = MyReader(System.`in`)
private fun next() = reader.next()
private fun nextInt() = next().toInt()
private fun nextLong() = next().toLong()
private fun nextDouble() = next().toDouble()
private fun nextStrings(n: Int) = Array(n) { next() }
private fun nextInts(n: Int) = IntArray(n) { nextInt() }
private fun nextLongs(n: Int) = LongArray(n) { nextLong() }
private fun nextDoubles(n: Int) = DoubleArray(n) { nextDouble() }